netdata/netdata
# eBPF mdflush

##### Summary

https://github.com/netdata/product/issues/1939

##### Component Name

ebpf.plugin

##### Test Plan

- Compile this branch.
- Check that the chart `mdstat.mdstat_flush` shows up.
- setup md device & filesystem on it, then do some random writes. use bpftool to see that the map contains flush details. see steps below:

```bash
$ dd if=/dev/zero of=/loopbackfile1.img bs=100M count=10
$ dd if=/dev/zero of=/loopbackfile2.img bs=100M count=10
$ dd if=/dev/zero of=/loopbackfile3.img bs=100M count=10
$ losetup /dev/loop50 /loopbackfile1.img
$ losetup /dev/loop51 /loopbackfile2.img
$ losetup /dev/loop52 /loopbackfile3.img
$ mdadm --create --verbose /dev/md1 /dev/loop50 /dev/loop51 /dev/loop52 --level=1 --raid-devices=3

# now create a partition table & ext4 filesystem on /dev/md1.
# mount /dev/md1p1 or whatever it is to some directory.
# write random files in the mounted area.
# see flushes start happening.
```

![image](https://user-images.githubusercontent.com/10018065/138068097-0c05b824-24cf-4892-844a-de3b530b26c9.png)

Tested with:

- Ubuntu 21.04 (Linux 5.11.0-31)
- TODO Ubuntu 18.04 (Linux 4.15)
- TODO CentOS 7 (Linux 3.10)
